SoftCCHierarchicalMNS.py

```diff
@@ -169,12 +173,15 @@ def det_module_subdir(self, ec):
                 tc_mpi_fullver = self.det_twodigit_version(tc_mpi)
                 subdir = os.path.join(MPI, tc_comp_name+tc_comp_ver, tc_mpi_name+tc_mpi_fullver)
 
-        if os.getenv('RSNT_ARCH') is None:
+        arch = os.getenv('RSNT_ARCH')
+        if arch is None:
             raise EasyBuildError("Need to set architecture to determine module path in $RSNT_ARCH")
-        if subdir != CORE and not subdir.startswith(os.path.join(CUDA, CUDA.lower())):
-            subdir = os.path.join(os.getenv('RSNT_ARCH'), subdir)
+        if using2023:
+            subdir = os.path.join('x86-64-v' + {'avx2': '3', 'avx512': '4'}[arch], subdir)
+        elif subdir != CORE and not subdir.startswith(os.path.join(CUDA, CUDA.lower())):
+            subdir = os.path.join(arch, subdir)
         elif tc_comp_name == GCCCORE.lower() and 'EBROOTGENTOO' in os.environ:
-            subdir = os.path.join(os.getenv('RSNT_ARCH'), subdir)
+            subdir = os.path.join(arch, subdir)
```

Comment:
This code could be simplified by always calculating: 
```
if using2023:
    arch_dir = 'x86-64-v' + {'avx2': '3', 'avx512': '4'}[arch]
else:
    arch_dir = arch
``` 

and then using `arch_dir` throughout without the need for `if using2023:`  (here and below) ?

Reply:
or, if it's separate functions, by implementing: 
``` 
def get_arch_dir(): 
     if 'EBROOTGENTOO' in os.environ and int(os.environ['EBVERSIONGENTOO']) >= 2023:
    arch_dir = 'x86-64-v' + {'avx2': '3', 'avx512': '4'}[arch]
else:
    arch_dir = arch
```

Reply:
I added the function but still need `using2023` to decide whether to use `arch_dir/Compiler/gcccore` and `arch_dir/Core` instead of `arch_dir/Core` and `Core`, respectively.